### PR TITLE
chore(dev-deps): update to jest 29 and TypeScript 4.8

### PR DESCRIPTION
**Pull request by Xhale1**
This PR pretty much just updates and cleans some dependencies, focusing on the dev experience. It should not effect the build of this library in a significant way.

- Jest 28 -> 29
- Remove ts-jest in favor of babel
- Update all deps to their minor versions
- Remove some unused deps (emotion-theming and @types/enzyme)
- Upgrade to typescript 4.8

closes: #388

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -183,11 +180,11 @@
     "react-16": "npm:react@16.14.0",
     "react-17": "npm:react@17.0.2",
     "react-dom": "18.2.0",
-    "react-dom-16": "npm:react-dom@16.14.0",
-    "react-dom-17": "npm:react-dom@17.0.2",
+    "react-dom-16": "npm:react-dom@^16.14.0",
+    "react-dom-17": "npm:react-dom@^17.0.2",
```
Comment (100terres) [inline package.json]: Based on our renovate configuration dev-deps are pinned.

```suggestion
    "react-dom-16": "npm:react-dom@16.14.0",
    "react-dom-17": "npm:react-dom@17.0.2",
```

**Review by 100terres — COMMENTED** [review]
Looks good to me :tada:

File: package.json (hunk 1 above)
Comment (Xhale1) [reply package.json]: Good catch, thanks

Comment (100terres) [reply package.json]: I'll do the fix and merge the PR :slightly_smiling_face:

**Review by 100terres — APPROVED** [review]

**Pull request merged**